firedrake/slate/slac/compiler.py

```diff
@@ -0,0 +1,324 @@
+                if not builder.needs_cell_facets:
+                    builder.require_cell_facets()
+                itsym = ast.Symbol("i0")
+                clist.append(ast.FlatBlock("&%s" % itsym))
+                loop_body = []
+                nfacet = exp.ufl_domain().ufl_cell().num_facets()
+
+                if integral_type == "exterior_facet":
+                    checker = 1
+                else:
+                    checker = 0
+                loop_body.append(ast.If(ast.Eq(ast.Symbol(cellfacetsym, rank=(itsym,)), checker),
+                                        [ast.Block([ast.FunCall(kinfo.kernel.name,
+                                                                tensor,
+                                                                coordsym,
+                                                                *clist)],
+                                                   open_scope=True)]))
+                loop = ast.For(ast.Decl("unsigned int", itsym, init=0), ast.Less(itsym, nfacet),
+                               ast.Incr(itsym, 1), loop_body)
+                statements.append(loop)
+            else:
+                statements.append(ast.FunCall(kinfo.kernel.name, tensor, coordsym, *clist))
+
+    # Now we handle any terms that require auxiliary data (if any)
+    if bool(builder.aux_exprs):
+        aux_temps, aux_statements = auxiliary_information(builder)
+        context_temps.update(aux_temps)
+        statements.extend(aux_statements)
+
+    result_sym = ast.Symbol("T%d" % len(builder.temps))
+    result_data_sym = ast.Symbol("A%d" % len(builder.temps))
+    result_type = "Eigen::Map<%s >" % eigen_matrixbase_type(shape)
+    result = ast.Decl(SCALAR_TYPE, ast.Symbol(result_data_sym, shape))
+    result_statement = ast.FlatBlock("%s %s((%s *)%s);\n" % (result_type,
+                                                             result_sym,
+                                                             SCALAR_TYPE,
+                                                             result_data_sym))
+    statements.append(result_statement)
+
+    cpp_string = ast.FlatBlock(metaphrase_slate_to_cpp(slate_expr, context_temps))
+    statements.append(ast.Assign(result_sym, cpp_string))
+
+    args = [result, ast.Decl("%s **" % SCALAR_TYPE, coordsym)]
+    for c in slate_expr.coefficients():
+        if isinstance(c, Constant):
+            ctype = "%s *" % SCALAR_TYPE
+        else:
+            ctype = "%s **" % SCALAR_TYPE
+        args.append(ast.Decl(ctype, builder.coefficient_map()[c]))
+
+    if builder.needs_cell_facets:
+        args.append(ast.Decl("char *", cellfacetsym))
+
+    macro_kernel_name = "compile_slate"
+    kernel_ast, oriented = builder.construct_ast(name=macro_kernel_name,
+                                                 args=args,
+                                                 statements=ast.Block(statements))
+
+    inc.append(petsc.get_petsc_dir() + '/include/eigen3/')
```

Comment:
With the `get_petsc_dir` change above, this should be something like:

```python
inc.extend(["%s/include/eigen3/" % d for d in get_petsc_dir()])
```

Probably set PETSC_DIR as an internal constant, so you only query it once.